modules/dataframe/src/properties/columnmetadatalistproperty.cpp

```diff
@@ -0,0 +1,145 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2021 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/dataframe/properties/columnmetadatalistproperty.h>
+#include <inviwo/core/metadata/metadataowner.h>
+#include <inviwo/core/datastructures/buffer/buffer.h>
+#include <inviwo/core/datastructures/buffer/bufferram.h>
+#include <inviwo/core/network/networklock.h>
+#include <inviwo/core/util/utilities.h>
+#include <inviwo/core/util/zip.h>
+#include <inviwo/core/util/stdextensions.h>
+#include <modules/base/algorithm/dataminmax.h>
+
+namespace inviwo {
+
+const std::string ColumnMetaDataListProperty::classIdentifier =
+    "org.inviwo.ColumnMetaDataListProperty";
+std::string ColumnMetaDataListProperty::getClassIdentifier() const { return classIdentifier; }
+
+ColumnMetaDataListProperty::ColumnMetaDataListProperty(std::string_view identifier,
+                                                       std::string_view displayName,
+                                                       InvalidationLevel invalidationLevel,
+                                                       PropertySemantics semantics)
+    : ListProperty(std::string(identifier), std::string(displayName),
```

Comment:
Why do you cast to string here? ListProperty should take string_views.

Reply:
forget this, seems I have not change this, at least not in master, and I can not figure out which brach It is in right now...